Ear karans contacts2

ear-dev opened the pull request:
This PR also includes Widechat changes to accomodate our toolbar.

TODO: 
disable public channels (maybe that's a server side setting?)
merge Karan's Kotlin SDK into ours in order to pass in CircleCI

kb0304 committed:
Add Floating Action Button

philipbrito committed:
Adds analytics for reset password event.

kb0304 committed:
WIP contacts

kb0304 committed:
Add the fragment transaction to contacts' list when FAB clicked

kb0304 committed:
Fix incorrect renames caused by autorefactoring tool

kb0304 committed:
Add create channel and invite contact buttons

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

kb0304 committed:
fix sort

kb0304 committed:
Fetch email for contacts, add search, add animations, back button, new activity

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

kb0304 committed:
UI improvements

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

philipbrito committed:
Merge branch 'develop' of github.com:RocketChat/Rocket.Chat.Android into new/analytics-for-password-reset

philipbrito committed:
Merge branch 'new/analytics-for-password-reset' of github.com:RocketChat/Rocket.Chat.Android into new/analytics-for-password-reset

kb0304 committed:
remove contact activity, use only fragment

kb0304 committed:
Reverte change to MainActivity, update toolbar icon in fragments

kb0304 committed:
set transition on fab click and create new channel click transaction

kb0304 committed:
Merge branch 'develop' of https://github.com/RocketChat/Rocket.Chat.Android into contacts_2

kb0304 committed:
Fix new chat toolbar

kb0304 committed:
Integrate invite methods from SDK

rafaelks committed:
Merge branch 'develop' into new/analytics-for-password-reset

kb0304 committed:
Merge branch 'develop' into contacts_2

kb0304 committed:
Update AndroidManifest.xml

kb0304 committed:
Update AndroidManifest.xml

kb0304 committed:
Update AndroidManifest.xml

kb0304 committed:
Update ContactsFragment.kt

philipbrito committed:
Merge branch 'develop' into new/analytics-for-password-reset

philipbrito committed:
Updates Analytics.kt

philipbrito committed:
Merge pull request #1790 from RocketChat/new/analytics-for-password-reset

[NEW] Adds analytics for reset password event.

ericrosenthal committed:
Merge branch 'contacts_2' into ear_karans_contacts2

ericrosenthal committed:
cleanup create channel toolbar

jaytat0 commented on app/src/main/java/chat/rocket/android/contacts/ContactsFragment.kt:
```diff
@@ -0,0 +1,322 @@
+    companion object {
+
+        /**
+         * Create a new ContactList fragment that displays the given list of contacts
```
This is actually ContactsFragment not ContactList fragment

ericrosenthal committed:
use magnifying glass instead of expanded search by default

jaytat0 commented on app/src/main/java/chat/rocket/android/contacts/models/Contact.kt:
```diff
@@ -0,0 +1,70 @@
+        dest?.writeString(name)
+        dest?.writeString(phoneNumber)
+    }
+
```
why do this function and the constructor above it not include email in the parcel?

ear-dev replied:
I have no idea.

I'm not sure how much we should be owning this PR since upstream will have to review.  Or... do we want to bring Karan into this review?

ericrosenthal committed:
add new strings to be translated

ear-dev committed:
Update README.md

ericrosenthal committed:
Merge branch 'veranda' into ear_karans_contacts2

ericrosenthal committed:
Merge branch 'develop' into ear_karans_contacts2

ericrosenthal committed:
Merge branch 'ear_karans_contacts2' of https://github.com/WideChat/Rocket.Chat.Android into ear_karans_contacts2

ear-dev committed:
Update README.md

ear-dev committed:
workaround for build issue

Caused by: com.android.tools.r8.utils.AbortException: Error: Class descriptor 'Landroid/support/customtabs/ICustomTabsCallback/;' cannot be represented in dex format.
